includes/data-port/models/class-sensei-import-lesson-model.php

```diff
@@ -0,0 +1,517 @@
+				'sensei_data_port_passmark_incorrect',
+				__( 'Both Passmark and Pass Required should be supplied.', 'sensei-lms' )
+			);
+		}
+
+		if ( null !== $pass_required ) {
+			if ( true === $pass_required ) {
+				$meta['_pass_required'] = 'on';
+				$meta['_quiz_passmark'] = $pass_mark;
+			} else {
+				$meta['_pass_required'] = '';
+				$meta['_quiz_passmark'] = 0;
+			}
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_NUM_QUESTIONS );
+		if ( null !== $value ) {
+			$meta['_show_questions'] = $value;
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_RANDOMIZE );
+		if ( null !== $value ) {
+			$meta['_random_question_order'] = true === $value ? 'yes' : 'no';
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_AUTO_GRADE );
+		if ( null !== $value ) {
+			$meta['_quiz_grade_type'] = true === $value ? 'auto' : 'manual';
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_QUIZ_RESET );
+		if ( null !== $value ) {
+			$meta['_enable_quiz_reset'] = true === $value ? 'on' : '';
+		}
+
+		return $meta;
+	}
+
+	/**
+	 * Helper method to parse and store the quiz questions.
+	 *
+	 * @param int $quiz_id  The quiz id.
+	 *
+	 * @return bool|WP_Error  True on success, WP_Error on failure.
+	 */
+	private function set_quiz_questions( $quiz_id ) {
+		$questions = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_QUESTIONS );
+		if ( null === $questions ) {
+			return true;
+		}
+
+		if ( empty( $questions ) ) {
+			$this->delete_quiz_question_meta( $quiz_id );
+			delete_post_meta( $this->get_post_id(), '_quiz_has_questions' );
+			delete_post_meta( $quiz_id, '_question_order' );
+
+			return true;
+		}
+
+		$question_import_ids = array_unique( Sensei_Data_Port_Utilities::split_list_safely( $questions, true ) );
+		$question_ids        = [];
+
+		foreach ( $question_import_ids as $question_import_id ) {
+			$question_id = $this->translate_import_id( Sensei_Data_Port_Question_Schema::POST_TYPE, $question_import_id );
+
+			if ( empty( $question_id ) ) {
+				return new WP_Error(
+					'sensei_data_port_question_not_found',
+					// translators: Placeholder is the term which errored.
+					sprintf( __( 'Question does not exist: %s.', 'sensei-lms' ), $question_id )
+				);
+			}
+
+			$question_ids[] = $question_id;
+		}
+
+		$old_question_order = get_post_meta( $quiz_id, '_question_order', true );
+		$old_question_order = empty( $old_question_order ) ? [] : array_map( 'intval', $old_question_order );
+
+		if ( $question_ids === $old_question_order ) {
+			return true;
+		}
+
+		$added_questions   = array_diff( $question_ids, $old_question_order );
+		$removed_questions = array_diff( $old_question_order, $question_ids );
+
+		// Delete question meta from the questions that were removed from the quiz.
+		if ( ! empty( $removed_questions ) ) {
+			$this->delete_quiz_question_meta( $quiz_id, array_diff( $old_question_order, $question_ids ) );
```

Comment:
This `array_diff()` could just use `$removed_questions`

Reply:
Oups! 87f4312f8ff9eca7c5a338e2ebe9efe5f3abc791